tests/PendingTenantsTest.php

```diff
@@ -0,0 +1,204 @@
+use Stancl\Tenancy\Tests\Etc\Tenant;
+
+test('tenants are correctly identified as pending', function (){
+    Tenant::createPending();
+
+    expect(Tenant::onlyPending()->count())->toBe(1);
+
+    Tenant::onlyPending()->first()->update([
+        'pending_since' => null
+    ]);
+
+    expect(Tenant::onlyPending()->count())->toBe(0);
+});
+
+test('pending trait adds query scopes', function () {
+    Tenant::createPending();
+    Tenant::create();
+    Tenant::create();
+
+    expect(Tenant::onlyPending()->count())->toBe(1)
+        ->and(Tenant::withPending(true)->count())->toBe(3)
+        ->and(Tenant::withPending(false)->count())->toBe(2)
+        ->and(Tenant::withoutPending()->count())->toBe(2);
+
+});
+
+test('pending tenants can be created and deleted using commands', function () {
+    config(['tenancy.pending.count' => 4]);
+
+    Artisan::call(CreatePendingTenants::class);
+
+    expect(Tenant::onlyPending()->count())->toBe(4);
+
+    Artisan::call(ClearPendingTenants::class);
+
+    expect(Tenant::onlyPending()->count())->toBe(0);
+});
+
+test('CreatePendingTenants command can have an older than constraint', function () {
+    config(['tenancy.pending.count' => 2]);
+
+    Artisan::call(CreatePendingTenants::class);
+
+    tenancy()->model()->query()->onlyPending()->first()->update([
+        'pending_since' => now()->subDays(5)->timestamp
+    ]);
+
+    Artisan::call('tenants:pending-clear --older-than-days=2');
+
+    expect(Tenant::onlyPending()->count())->toBe(1);
+});
+
+test('CreatePendingTenants command cannot run with both time constraints', function () {
+    pest()->artisan('tenants:pending-clear --older-than-days=2 --older-than-hours=2')
+        ->assertFailed();
+});
+
+test('CreatePendingTenants commands all option overrides any config constraints', function () {
+    Tenant::createPending();
+    Tenant::createPending();
+
+    tenancy()->model()->query()->onlyPending()->first()->update([
+        'pending_since' => now()->subDays(10)
+    ]);
+
+    config(['tenancy.pending.older_than_days' => 4]);
+
+    Artisan::call(ClearPendingTenants::class, [
+        '--all' => true
+    ]);
+
+    expect(Tenant::onlyPending()->count())->toBe(0);
+});
+
+test('tenancy can check for pending tenants', function () {
+    Tenant::query()->delete();
+
+    expect(Tenant::onlyPending()->exists())->toBeFalse();
+
+    Tenant::createPending();
+
+    expect(Tenant::onlyPending()->exists())->toBeTrue();
+});
+
+test('tenancy can pull a pending tenant', function () {
+    expect(Tenant::pullPendingTenant())->toBeNull();
+
+    Tenant::createPending();
+
+    expect(Tenant::pullPendingTenant(true))->toBeInstanceOf(Tenant::class);
+});
+
+test('tenancy can create if none are pending', function () {
+    expect(Tenant::all()->count())->toBe(0);
+
+    Tenant::pullPendingTenant(true);
+
+    expect(Tenant::all()->count())->toBe(1);
+});
+
+test('the include in queries global scope can include pending tenants in all queries', function () {
```

Comment:
```suggestion
test('pending tenants are included in all queries depending on the value of the include in queries config', function () {
```

@stancl I think I'd change the test name to this since we're testing if changing the config value makes queries include the pending tenants. Also, "include in queries global scope" doesn't seem accurate since the scope is named "PendingScope"

Reply:
Other than this, the test names seem fine

Reply:
Agree that `include in queries` is not exactly the global scope, but the original sentence seems to have a better structure.

Also in your suggested change: `are included ... depending on` doesn't make sense, it should be `can be included ... depending on` (or `based on`)

Also I wouldn't go too verbose here, `the value of the` can be omitted.

Reply:
I see

```suggestion
test('pending tenants are included in queries based on the include_in_queries config', function () {
```

I also omitted the `all` from `all queries` (I think `queries` already implies `all queries`), and I added underscores so that it's clear we're talking about a config key (not sure about having special characters in the test names, but the tests run fine, so I think it's OK to keep the underscores). What do you think about the new suggestion?

Reply:
> I think queries already implies all queries

The `all` is the key emphasis here, so I'd keep it. That's the point of the option — instead of including the tenants only in *some* queries, they are included in *all* queries.

And yeah, I omitted underscores initially but I think we can use any characters in test names now that we're using pest.

Reply:
> That's the point of the option — instead of including the tenants only in some queries, they are included in all queries

Right. I'll edit the suggestion and commit it